### PR TITLE
Gnome45

### PR DESCRIPTION
**Pull request by gilvbp**
Partially working, //TODO need to fix some features 

Fixes Issue: https://github.com/RaphaelRochet/arch-update/issues/216
image

### CODE REVIEW

**Review by RaphaelRochet — CHANGES_REQUESTED** [review]
Thanks for the effort, much appreciated.
But when you change every bit of space in code to your liking, if create so much mess in the diff that's not readable.  Please respect the current code spacing. And do not change anything that's not related to the PR.

**Comment by RaphaelRochet** [comment]
That's going to be interesting ...

File: .eslintrc.yml
[the remark this answers is not in the document]
Comment (gilvbp) [reply .eslintrc.yml:1]: sorry, I will remove

File: prefs.js [hunk 1]
```diff
@@ -16,62 +16,58 @@
...
+        settings.bind('update-cmd', buildable.get_object('field_updatecmd'), 'text', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('pacman-dir', buildable.get_object('field_pacmandir'), 'text', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('auto-expand-list', buildable.get_object('field_autoexpandlist'), 'value', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('package-manager', buildable.get_object('field_packagemanager'), 'text', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('enable-positioning', buildable.get_object('field_enablepositioning'), 'active', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('enable-positioning', buildable.get_object('box_position'), 'sensitive', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('position', buildable.get_object('field_position'), 'active', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('position-number', buildable.get_object('field_positionnumber'), 'value', Gio.SettingsBindFlags.DEFAULT);
+        settings.bind('package-info-cmd', buildable.get_object('field_packageinfocmd'), 'text', Gio.SettingsBindFlags.DEFAULT);
```
[the remark this answers is not in the document]
Comment (gilvbp) [reply prefs.js:65]: > Oh, you didn't like my spacings ? What a mess. I get the `Me` to `this` change though.

My mistake. I'll rollback the code format. That was intellij

File: metadata.json [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
   "name": "Arch Linux Updates Indicator",
-  "shell-version": [ "44" ],
-  "url": "https://github.com/RaphaelRochet/arch-update", 
-  "uuid": "arch-update@RaphaelRochet", 
-  "version": 53
+  "shell-version": [ "45" ],
+  "url": "https://github.com/RaphaelRochet/arch-update",
+  "uuid": "arch-update@RaphaelRochet",
+  "gettext-domain": "arch-update",
+  "settings-schema": "org.gnome.shell.extensions.arch-update",
```
[the remark this answers is not in the document]
Comment (gilvbp) [reply metadata.json:8]: Yes, i just remember to eliminate the initTranslation method. Check this: https://gjs.guide/extensions/upgrading/gnome-shell-45.html#extensionutils
Just fix with this commit: https://github.com/gilvbp/arch-update/commit/b1d144c242ec4ab4870f65892b3766317d6fffe6

**Review by gilvbp — COMMENTED** [review]
Review finished

File: extension.js [hunk 3]
```diff
@@ -17,584 +17,614 @@
...
+                        // We have to prepare this only once
+                        this._notifSource = new MessageTray.SystemNotificationSource();
+                        this._notifSource.createIcon = function () {
+                            let gicon = Gio.icon_new_for_string(`${archUpdateIndicatorExtension.dir.get_child('icons').get_path()}/arch-updates-logo.svg`);
+                            return new St.Icon({gicon});
+                        };
+                        // Take care of note leaving unneeded sources
+                        this._notifSource.connect('destroy', () => {
+                            this._notifSource = null;
+                        });
```
Comment (RaphaelRochet) [inline extension.js:591]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 4]
```diff
@@ -17,584 +17,614 @@
...
+                    this._updateProcess_sourceId = null;
+                    this._updateProcess_pid = null;
+                    // Update indicator
+                    this._showChecking(false);
+                    if (DISABLE_PARSING) {
+                        this._updateStatus(this._updateList.length);
+                    } else {
+                        this._updateStatus(this._updateList.filter(function (line) {
+                            return RE_UpdateLine.test(line);
+                        }).length);
```
Comment (RaphaelRochet) [inline extension.js:576]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 5]
```diff
@@ -17,584 +17,614 @@
...
+                }
+
+                _checkUpdatesRead() {
+                    // Read the buffered output
+                    let updateList = [];
+                    let out, size;
+                    do {
+                        [out, size] = this._updateProcess_stream.read_line_utf8(null);
+                        if (out)
+                            updateList.push(out);
```
Comment (RaphaelRochet) [inline extension.js:556]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 6]
```diff
@@ -17,584 +17,614 @@
...
+                    } catch (err) {
+                        this._showChecking(false);
+                        this.lastUnknowErrorString = err.message.toString();
+                        this._updateStatus(-2);
+                    }
+                }
+
+                _cancelCheck() {
+                    if (this._updateProcess_pid == null)
+                        return;
```
Comment (RaphaelRochet) [inline extension.js:541]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 7]
```diff
@@ -17,584 +17,614 @@
...
+
+
+                    Util.spawnCommandLine(`kill ${this._updateProcess_pid}`);
```
Comment (RaphaelRochet) [inline extension.js:544]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 8]
```diff
@@ -17,584 +17,614 @@
...
+                        }).length);
+                    }
+                }
+
+                _showNotification(title, message) {
+                    if (this._notifSource == null) {
+                        // We have to prepare this only once
+                        this._notifSource = new MessageTray.SystemNotificationSource();
+                        this._notifSource.createIcon = function () {
+                            let gicon = Gio.icon_new_for_string(`${archUpdateIndicatorExtension.dir.get_child('icons').get_path()}/arch-updates-logo.svg`);
```
Comment (RaphaelRochet) [inline extension.js:585]: Don't change existing code without good reason, and related to PR
I get the change from Me to archUpdateIndicatorExtension though.

File: extension.js [hunk 9]
```diff
@@ -17,584 +17,614 @@
...
+                        Main.messageTray.add(this._notifSource);
+                    }
+                    let notification = null;
+                    // We do not want to have multiple notifications stacked
+                    // instead we will update previous
+                    if (this._notifSource.notifications.length == 0) {
+                        notification = new MessageTray.Notification(this._notifSource, title, message);
+                        notification.addAction(_('Update now'), () => {
+                            this._updateNow();
+                        });
```
Comment (RaphaelRochet) [inline extension.js:601]: Don't change existing code without good reason, and related to PR

File: extension.js [hunk 10]
```diff
@@ -17,584 +17,614 @@
...
+    }
+
+    init() {
+        String.prototype.format = String.prototype.f = function () {
+            var s = this, i = arguments.length;
+            while (i--)
+                s = s.replace(new RegExp(`\\{${i}\\}`, 'gm'), arguments[i]);
+            return s;/* www .  j  a va 2 s . c  o m*/
+        };
+    }
```
Comment (RaphaelRochet) [inline extension.js:619]: What is that ?

**Review by RaphaelRochet — CHANGES_REQUESTED** [review]
With all your unrelated changes (spacing, templates, empty lines, refactoring) you make it very very hard to review changes.
Please review all your changes so that there is only changes strictly required and related to the PR.
Also please respect existing code style : spacings, empty lines, strings ...

**Comment by RaphaelRochet** [comment]
Also can you tell what is missing ?
I may end up creating a new branch, if this is not fully functional.

**Comment by gilvbp** [comment]
> Also can you tell what is missing ? I may end up creating a new branch, if this is not fully functional

It's better to create a new branch code (ex gnome-45), and I will rollback some code, eslintrct and IntelliJ formated the code that messed up. Need to rollback. Prefs.js and Update list button need to get working.

**Pull request closed without merging**